Source/Project64/N64 System/Mips/Pif Ram.cpp

```diff
@@ -297,19 +297,16 @@ void CPifRam::SI_DMA_READ()
 			{
 				continue;
 			}
-			RDRAM[RdramPos ^3] = m_PifRam[count];
+			RDRAM[BES(RdramPos)] = m_PifRam[count];
 		}
 	}
 	else
 	{
-		for (size_t i = 0; i < 64; i += 4)
-		{
-			unsigned __int32 pif_ram_dword;
-			std::memcpy(&pif_ram_dword, &PifRamPos[i], sizeof(unsigned __int32));
-
-			pif_ram_dword = swap32by8(pif_ram_dword);
+		register size_t i;

```

Comment:
I also suspect that, for this function, the type of `i` has no reason to be different between each of these `for` loops in the cleaned-up end result, so while your reasoning is that `i` should not be declared outside of the relevant local scope, I am sure that the declaration can be re-used for multiple `for` loops for easier maintainability over changing what the loop variable type is in a single declaration, not multiple repeated super-local declarations.